### PR TITLE
feat: add kpi calc for tether campaign

### PR DESCRIPTION
**Pull request by kathaypacific**
As the title.

Related to ENG-495

### CODE REVIEW

File: scripts/calculateKpi/protocols/tetherV0/index.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          transactionHashesCounted.add(transactionHash)
+        }
+      }
+    }
+  })
+
+  return totalTransactions
+}
+
+export async function calculateKpi({
```
Comment (jeanregisser) [inline scripts/calculateKpi/protocols/tetherV0/index.ts:108]: nit: we should document this function like we did for the other campaigns (KPI unit, business purpose, calculation method, etc).

File: scripts/calculateKpi/protocols/tetherV0/index.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  // Since we are using events to count transfers, we need to ensure that transactions that emit multiple events are only counted once
+  const transactionHashesCounted = new Set<string>()
+  let totalTransactions = 0
+
+  const query = {
+    transactions: [{ from: [user] }],
+    logs: [
+      {
+        contractAddress: tokenAddress,
+        topics: [[transferEventSigHash], [pad(user, { size: 32 })], [], []],
```
Comment (jeanregisser) [inline scripts/calculateKpi/protocols/tetherV0/index.ts:61]:
```suggestion
        contractAddress: tokenAddress,
        // transfer from user
        topics: [[transferEventSigHash], [pad(user, { size: 32 })], [], []],
```

File: scripts/calculateKpi/protocols/tetherV0/index.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      },
+      {
+        contractAddress: tokenAddress,
+        topics: [[transferEventSigHash], [], [pad(user, { size: 32 })], []],
```
Comment (jeanregisser) [inline scripts/calculateKpi/protocols/tetherV0/index.ts:65]:
```suggestion
        contractAddress: tokenAddress,
        // transfer to user
        topics: [[transferEventSigHash], [], [pad(user, { size: 32 })], []],
```

File: scripts/calculateKpi/protocols/tetherV0/index.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+      if (data && transactionHash) {
+        const decodedLog = decodeEventLog({
+          abi: erc20Abi,
+          data: data as Hex,
+          topics: topics as [],
+        })
+        const transferValue = decodedLog.args.value
+
+        if (
+          BigNumber(transferValue).gt(MIN_ELIGIBLE_VALUE_IN_SMALLEST_UNIT) &&
```
Comment (jeanregisser) [inline scripts/calculateKpi/protocols/tetherV0/index.ts:95]: right?
```suggestion
          BigNumber(transferValue).gte(MIN_ELIGIBLE_VALUE_IN_SMALLEST_UNIT) &&
```

File: scripts/calculateKpi/protocols/tetherV0/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+          {
+            data: encodedValueAboveThreshold,
+            topics: [
+              transferEventSigHash,
+              pad(testAddress, { size: 32 }),
+              pad('0x4567890123456789012345678901234567890123' as Address, {
+                size: 32,
+              }),
+              '0x0000000000000000000000000000000000000000000000000000000000000000',
+            ],
```
Comment (jeanregisser) [inline scripts/calculateKpi/protocols/tetherV0/index.test.ts:128]: supernit: we could maybe consider using `encodeEventTopics` and `encodeAbiParameters` from viem to make it more readable?

Something similar to what we did in https://github.com/divvi-xyz/divvi-relayer/blob/a55f8fc95a8b16547336e22207e4af0f23b324a2/src/services/divviRegistry.test.ts#L98-L113

File: scripts/calculateKpi/protocols/tetherV0/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+            transactionHash: '0xabc123', // Same transaction hash
+          },
+        ])
+        await onPage(mockResponse)
+      })
+
+      const result = await calculateKpi(defaultProps)
+
+      // Should count as 1 transaction per network, not 2
```
Comment (jeanregisser) [inline scripts/calculateKpi/protocols/tetherV0/index.test.ts:197]: Edge casey: should we calculate the net value transfer for the user?
And accept a transaction that come with more than 1 transfer to reach the minimum value?

**Review by jeanregisser — APPROVED** [review]
Awesome! 🚀

File: scripts/calculateKpi/protocols/tetherV0/index.test.ts (hunk 6 above)
Comment (kathaypacific) [reply scripts/calculateKpi/protocols/tetherV0/index.test.ts:197]: this is a great point, i've updated the logic to count net transfers per tx!
